src/backend/EinsumNode.h

```diff
@@ -79,6 +80,26 @@ class einsum_ir::backend::EinsumNode {
     //! binary contraction
     BinaryContraction * m_cont = nullptr;
 
+    //! Memory manager for intermendiate results
+    MemoryManager * m_memory = nullptr;
+
+    //! id of allocated memoy
+    int64_t m_mem_id = 0;
+
+    //! the required number of frees until reserved memory is actualy freed
+    int64_t m_req_mem_frees = 1;
+
+    //! the number of reserved memory frees 
+    int64_t m_num_mem_frees = 0;
```

Comment:
What exactly is counted here? I am assuming that this is some sort of bookkeeping signaling that they don't need their data anymore?
Maybe there is a better name for the variable?